lib/src/parser/recursive.dart

```diff
@@ -531,6 +539,111 @@ class RecursiveAstParser {
+      NgToken innerValue) {
+    var text = innerValue.lexeme;
+    var absoluteTextOffset = innerValue.offset;
+    var tokens = new RegExp(r'({{)|(}})');
+    var mustaches = <InterpolationAst>[];
+    int seenOpenMustache;
+
+    var scanner = new StringScanner(text);
+    while (!scanner.isDone) {
+      var position = scanner.position;
+      var match = tokens.firstMatch(scanner.rest);
+      if (match != null) {
+        var matchPosition = position + match.start;
+
+        if (match.group(1) != null) {
+          // '{{' found.
+          if (seenOpenMustache == null) {
+            seenOpenMustache = matchPosition;
+          } else {
+            // Second '{{' found before '}}' closes it.
+            var firstMustacheBegin = absoluteTextOffset + seenOpenMustache;
+            exceptionHandler.handle(new AngularParserException(
+              NgParserWarningCode.UNTERMINATED_MUSTACHE,
+              firstMustacheBegin,
+              '{{'.length,
+            ));
+
+            mustaches.add(new InterpolationAst.parsed(
+              _source,
+              new NgToken.interpolationStart(firstMustacheBegin),
+              new NgToken.interpolationValue(
+                  firstMustacheBegin + '{{'.length,
+                  text.substring(
+                      seenOpenMustache + '{{'.length, matchPosition)),
+              new NgToken.generateErrorSynthetic(
+                  absoluteTextOffset + matchPosition,
+                  NgTokenType.interpolationEnd),
+            ));
+            seenOpenMustache = matchPosition;
+          }
+        }
+
+        if (match.group(2) != null) {
+          if (seenOpenMustache != null) {
+            var mustacheBegin = absoluteTextOffset + seenOpenMustache;
+            var mustacheEnd = absoluteTextOffset + matchPosition;
+            mustaches.add(new InterpolationAst.parsed(
+              _source,
+              new NgToken.interpolationStart(mustacheBegin),
+              new NgToken.interpolationValue(
+                  mustacheBegin + '{{'.length,
+                  text.substring(
+                      seenOpenMustache + '{{'.length, matchPosition)),
+              new NgToken.interpolationEnd(mustacheEnd),
+            ));
+            seenOpenMustache = null;
+          } else {
+            // Found '}}' before any '{{'
+            var mustacheBegin = absoluteTextOffset + position;
+            var mustacheEnd = absoluteTextOffset + matchPosition;
+            exceptionHandler.handle(new AngularParserException(
+              NgParserWarningCode.UNOPENED_MUSTACHE,
+              mustacheEnd,
+              '}}'.length,
+            ));
+            mustaches.add(new InterpolationAst.parsed(
+              _source,
+              new NgToken.generateErrorSynthetic(
+                  mustacheBegin, NgTokenType.interpolationStart),
+              new NgToken.interpolationValue(
+                  mustacheBegin, text.substring(position, matchPosition)),
```

Comment:
Interesting.

We had this same problem in our code, and I remember that our first approach (assuming everything after {{ is intentionally a binding so we could autocomplete) caused problems when there was a space. The issue was that `{{ Bob` was tokenized and parsed, but the tokenizer drops whitespace. So it was treated as `{{ [DART_HERE]Bob[/DART_HERE]`. This meant that the `{{` was completely detached from the recognized dart code, and wasn't autocompletable. We changed it to only parse when its not followed by a space ie `{{Bob`.

The dart team said what we should do is make a synthetic parenthesized expression so that the dart code can contain the whitespace. If we do this, we don't need to worry about spaces. But if we don't, we do need to worry about those spaces preventing autocomplete.